### PR TITLE
[Windows] Slight rework of the PyPy installation

### PR DESCRIPTION
**Pull request by shilovmaksim**
# Description
Avoid parsing PyPy downloads "html" page. Use versions config provided by PyPy developers instead, which is considered to be more stable.

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/Installers/Install-PyPy.ps1 [hunk 1]
```diff
@@ -83,35 +67,29 @@ function Install-PyPy
...
-            $tempPyPyPackagePath = Start-DownloadWithRetry -Url $url -Name  $packageName
-            Install-PyPy -PackagePath $tempPyPyPackagePath -Architecture $pypyTool.arch
-        }
-        else
-        {
-            Write-Host "Failed to query PyPy version '$pypyVersion' by '$filter' filter"
-            exit 1
-        }
+        Write-Host "Found PyPy '$($latestMajorPyPyVersion.filename)' package"
+        $tempPyPyPackagePath = Start-DownloadWithRetry -Url $latestMajorPyPyVersion.download_url -Name  $latestMajorPyPyVersion.filename
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-PyPy.ps1:87]:
```suggestion
        $tempPyPyPackagePath = Start-DownloadWithRetry -Url $latestMajorPyPyVersion.download_url -Name $latestMajorPyPyVersion.filename

```

Comment (shilovmaksim) [reply images/win/scripts/Installers/Install-PyPy.ps1:87]: Sigh

File: images/win/scripts/Installers/Install-PyPy.ps1 [hunk 2]
```diff
@@ -83,35 +67,29 @@ function Install-PyPy
...
 Write-Host "Starting installation PyPy..."
-foreach($pypyTool in $pypyTools)
+foreach($toolsetVersion in $toolsetVersions.versions)
 {
-    foreach($pypyVersion in $pypyTool.versions)
+    # Query latest PyPy version
+    $latestMajorPyPyVersion = ($pypyVersions |
+        Where-Object {$_.python_version.StartsWith("$toolsetVersion") -and $_.stable -eq $true} |
+        Select-Object -First 1).files |
+        Where-Object platform -like "win??"
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-PyPy.ps1:82]:
```suggestion
        Where-Object platform -like "win*"

```

Comment (shilovmaksim) [reply images/win/scripts/Installers/Install-PyPy.ps1]: Done

File: images/win/scripts/Installers/Install-PyPy.ps1 (hunk 1 above)
Comment (shilovmaksim) [reply images/win/scripts/Installers/Install-PyPy.ps1]: Done

**Review by miketimofeev — APPROVED** [review]

**Review by miketimofeev — DISMISSED** [review]

File: images/win/scripts/Installers/Install-PyPy.ps1 [hunk 3]
```diff
@@ -83,35 +67,29 @@ function Install-PyPy
...
 
 Write-Host "Starting installation PyPy..."
-foreach($pypyTool in $pypyTools)
+foreach($toolsetVersion in $toolsetVersions.versions)
 {
-    foreach($pypyVersion in $pypyTool.versions)
+    # Query latest PyPy version
+    $latestMajorPyPyVersion = ($pypyVersions |
+        Where-Object {$_.python_version.StartsWith("$toolsetVersion") -and $_.stable -eq $true} |
+        Select-Object -First 1).files |
```
Comment (al-cheb) [inline images/win/scripts/Installers/Install-PyPy.ps1:81]: Let's move .files to  `-ExpandProperty files` param and remove ()

Comment (shilovmaksim) [reply images/win/scripts/Installers/Install-PyPy.ps1]: Done

**Review by al-cheb — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Pull request merged**